Helper/TrackingSaveObserver.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Bolt\Boltpay\Observer;
+
+use Bolt\Boltpay\Helper\Api as ApiHelper;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Magento\Framework\DataObjectFactory;
+
+/**
+ * Class TrackingSaveObserver
+ *
+ * @package Bolt\Boltpay\Observer\Adminhtml\Sales
+ */
+class TrackingSaveObserver implements ObserverInterface
+{
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+
+    /**
+     * @var DataObjectFactory
+     */
+    private $dataObjectFactory;
+
+    /**
+     * @param ConfigHelper      $configHelper
+     * @param DataObjectFactory $dataObjectFactory
+     *
+     */
+    public function __construct(
+        ConfigHelper $configHelper,
+        DataObjectFactory $dataObjectFactory
+    ) {
+        $this->configHelper = $configHelper;
+        $this->dataObjectFactory = $dataObjectFactory;
+    }
+
+    /**
+     * @param Observer $observer
+     */
+    public function execute(Observer $observer)
+    {
+        $tracking = $observer->getEvent()->getTrack();
+        $shipment = $tracking->getShipment();
+        $order = $shipment->getOrder();
+        $payment = $order->getPayment();
+        $transactionReference = $payment->getAdditionalInformation('transaction_reference');
```

Comment:
This appears to work but I have the feeling there should be an easier way to work backwards to the transaction reference from here. Is there a helper I'm forgetting @danac-gs @YevhenShBolt ?

If there isn't perhaps we should create it.

Reply:
Not sure there's a shorter path to transaction_reference in this case.